source/libraries/CompileSolidity.ts

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as solc from 'solc';
+
+class SolidityContractCompiler {
+    private compiler: any = solc;
+    private contractInputDirectoryPath: string;
+    private contractOutputDirectoryPath: string;
+    private contractOutputFileName: string;
+
+    public constructor(contractInputDirectoryPath: string, contractOutputDirectoryPath: string, contractOutputFileName: string) {
+        this.contractInputDirectoryPath = contractInputDirectoryPath;
+        this.contractOutputDirectoryPath = contractOutputDirectoryPath;
+        this.contractOutputFileName = contractOutputFileName;
+    }
+
+    public compileContracts() {
+        // Compile all contracts in this.inputDirectoryPath
+        let inputJSON: any = this.generateCompilerInput();
+        let compilerOutput: any = this.compiler.compile(inputJSON);
+        if (compilerOutput.errors.length > 0) {
+            console.log("The compiler encountered the following warnings/errors:");
+            for (let error of compilerOutput.errors) {
+                console.log(error);
+            }
+            // Do not return automatically in case there are only warnings and no errors
+        }
+
+        // Create output directory (if it doesn't exist) and save contract bytecodes to single file
```

Comment:
The output will include more than just bytecode, it will also include ABI, metadata, potentially the AST, etc.  You can configure all of this via the input JSON, but for now lets go with defaults and see what we get.